beacon-chain/blockchain/process_block.go

```diff
@@ -528,7 +524,9 @@ func (s *Service) lateBlockTasks(ctx context.Context) {
 	if err := transition.UpdateNextSlotCache(ctx, lastRoot, lastState); err != nil {
 		log.WithError(err).Debug("could not update next slot state cache")
 	}
-
+	if err := s.handleEpochBoundary(ctx, currentSlot, headState, headRoot[:]); err != nil {
```

Comment:
There is an edge case here, if `lastState=nil` then our head state becomes the last state. Since we also advance this particular state in `UpdateNextSlotCache` , we will now be advancing this again in `handleEpochBoundary` . We don't necessarily need to address it as lastState being nil should only happen at startup afaik. But something to  be aware of